src/backend/marsha/core/serializers/video.py

```diff
@@ -434,6 +434,128 @@ def validate(self, attrs):
         return super().validate(attrs)
 
 
+class SharedLiveMediaSerializer(
+    UploadableFileWithExtensionSerializerMixin, serializers.ModelSerializer
+):
+    """Serializer to display a shared live media model."""
+
+    class Meta:  # noqa
+        model = SharedLiveMedia
+        fields = (
+            "active_stamp",
+            "filename",
+            "id",
+            "is_ready_to_show",
+            "nb_pages",
+            "show_download",
+            "title",
+            "upload_state",
+            "urls",
+            "video",
+        )
+        read_only_fields = (
+            "active_stamp",
+            "filename",
+            "id",
+            "is_ready_to_show",
+            "nb_pages",
+            "upload_state",
+            "urls",
+            "video",
+        )
+
+    active_stamp = TimestampField(
+        source="uploaded_on", required=False, allow_null=True, read_only=True
+    )
+    filename = serializers.SerializerMethodField()
+    is_ready_to_show = serializers.BooleanField(read_only=True)
+    urls = serializers.SerializerMethodField()
+
+    def create(self, validated_data):
+        """Force the video field to the video of the JWT Token if any.
+
+        Parameters
+        ----------
+        validated_data : dictionary
+            Dictionary of the deserialized values of each field after validation.
+
+        Returns
+        -------
+        dictionary
+            The "validated_data" dictionary is returned after modification.
+
+        """
+        # user here is a video as it comes from the JWT
+        # It is named "user" by convention in the `rest_framework_simplejwt` dependency we use.
+        user = self.context["request"].user
+        # Set the video field from the payload if there is one and the user is identified
+        # as a proper user object through access rights
+        if (
+            self.initial_data.get("video")
+            and user.token.get("user")
+            and user.token["resource_id"] == user.token.get("user", {}).get("id")
+        ):
+            validated_data["video_id"] = self.initial_data.get("video")
+        # If the user just has a token for a video, force the video ID on the shared live media
+        if not validated_data.get("video_id") and isinstance(user, TokenUser):
+            validated_data["video_id"] = user.id
+        return super().create(validated_data)
+
+    def get_filename(self, obj):
+        """Filename of the shared live media."""
+        return (
+            self._get_filename(obj.title, extension=obj.extension)
+            if obj.uploaded_on
+            else None
+        )
```

Comment:
IMHO, this could be a property on the model.

Reply:
It can be it's out of the scope of this PR, this is not used only by this serializer but by other so it should be a more global refactor with all concerned models.